### PR TITLE
DAOS-9902 API: To support empty container lable

### PR DESCRIPTION
**Pull request by wiliamhuang**
Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

File: src/common/mfu_daos.c [hunk 1]
```diff
@@ -3857,10 +3857,12 @@ static int cont_serialize_prop_str(struct hdf5_args* hdf5,
...
-    if (entry == NULL || entry->dpe_str == NULL) {
-        MFU_LOG(MFU_LOG_ERR, "Property %s not found", prop_str);
-        rc = 1;
-        goto out;
+    if (entry == NULL || entry->dpe_str == NULL ) {
+        if ( !(entry != NULL && entry->dpe_str == NULL && entry->dpe_type == DAOS_PROP_CO_LABEL)) {
+            MFU_LOG(MFU_LOG_ERR, "Property %s not found", prop_str);
+            rc = 1;
+            goto out;
+        }
```
Comment (daltonbohning) [inline src/common/mfu_daos.c:3865]: Really, this was an oversight during original implementation. If DAOS gives us a NULL/empty string for ANY property, then we just trust that. The client code here doesn't need to enforce any business rules. So, we could simplify to something like...
```suggestion
    if (entry == NULL) {
        MFU_LOG(MFU_LOG_ERR, "Property %s not found", prop_str);
        rc = 1;
        goto out;
```

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/common/mfu_daos.c (hunk 1 above)
Comment (wiliamhuang) [reply src/common/mfu_daos.c]: Thank you! Just made the change you suggested.

**Review by daltonbohning — APPROVED** [review]

**Comment by daltonbohning** [comment]
@adammoody I went ahead and landed since the changes are contained to `mfu_daos.c`. In general, are you okay with this, as long DAOS changes are contained to DAOS-only source files?

**Comment by adammoody** [comment]
Yes, that's fine to go ahead a merge daos-specific items.  Thanks for checking, @daltonbohning .

**Pull request merged**